### ISSUE TITLE
handling gbk files with leader sequence

### ISSUE THREAD

**Issue by adityabandla** [antismash/antismash#218]
**Describe the bug**
@kblin I have been following your suggestion of using secmet to extract info from the genbank files. See #216. This works for most files, but fails for files which have the string `leader`, for example: ```prepeptide=leader, leader_location="[56991:57051](-)```

For such files, I get the error 
```
antismash.common.secmet.errors.SecmetInvalidInputError: Features that bridge the record origin cannot be directly created: join{[56991:57051](-)}
```

**Expected behavior**
These files should be parsed like others which don't have this string
Attached is a file which can help reproduce this error
[CS1EDSR2D_564265.region001.gbk.txt](https://github.com/antismash/antismash/files/4027635/CS1EDSR2D_564265.region001.gbk.txt)

**Comment by SJShaw** [comment]
Thanks for letting us know.

The issue isn't the leader section itself, but that the genbank files for regions don't properly shift the leader location qualifier, which can lead to locations being outside the bounds of the record or in an inverted order that is interpreted as being a feature crossing the origin in a circular genome.

While this is being fixed, you can use secmet to load the full genbank output and just operate on the relevant region within the record.

**Comment by adityabandla** [comment]
Thanks for the heads up. Python is not my language of choice and i'm just looking to get some basic summary info from the genbank files. I'm trying to pull out the record ID/contig ID, region length, product string and the CDS identifiers for the region

I was able to get it using the code you had shared previously, but i'm not entirely sure what changes are required here to get the CDS identifiers from the file described above
```
for record in Record.from_genbank(filename, taxon="bacteria"):  # or fungi, if relevant
    for region in record.get_regions():
        for cds in region.cds_children:
            print(record.id, region.get_product_string(), cds.get_name())
```

**Comment by SJShaw** [comment]
The `cds.get_name()` is the CDS identifier, using the first defined qualifier from `locus_tag`, `gene`, and then `protein_id`.

**Comment by adityabandla** [comment]
I checked using this commit (just replaced the two files manually), but still get the same error
```
raceback (most recent call last):
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/record.py", line 743, in from_biopython
    record.add_biopython_feature(feature)
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/record.py", line 623, in add_biopython_feature
    self.add_cds_motif(Prepeptide.from_biopython(feature, record=self))
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/features/prepeptide.py", line 225, in from_biopython
    alt_weights, leader, tail
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/features/prepeptide.py", line 46, in __init__
    super().__init__(location, locus_tag, protein_location=FeatureLocation(0, len(location)), tool=tool)
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/features/cds_motif.py", line 26, in __init__
    tool=tool, created_by_antismash=True)
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/features/domain.py", line 60, in __init__
    super().__init__(location, feature_type, tool=tool, created_by_antismash=created_by_antismash)
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/features/antismash_feature.py", line 26, in __init__
    super().__init__(location, feature_type, created_by_antismash=created_by_antismash)
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/features/feature.py", line 74, in __init__
    raise ValueError("Features that bridge the record origin cannot be directly created: %s" % location)
ValueError: Features that bridge the record origin cannot be directly created: join{[10895:10952](+), [4797:4926](+)}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "extract_as_regions.py", line 41, in <module>
    main()
  File "extract_as_regions.py", line 27, in main
    for record in Record.from_genbank(f, taxon="bacteria"): 
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/record.py", line 774, in from_genbank
    records.append(Record.from_biopython(bio, taxon))
  File "/home/projects/11001755/miniconda3/envs/antismash/lib/python3.7/site-packages/antismash/common/secmet/record.py", line 745, in from_biopython
    raise SecmetInvalidInputError(str(err)) from err
antismash.common.secmet.errors.SecmetInvalidInputError: Features that bridge the record origin cannot be directly created: join{[10895:10952](+), [4797:4926](+)}
```

**Comment by SJShaw** [comment]
That particular region file was generated incorrectly, and that generation is what's been fixed. You'll need to rerun antiSMASH on the original input to generate a region file that's correct.

**Comment by kblin** [comment]
Reopening until this fix is in a release.

**Comment by SJShaw** [comment]
Closing again, as the fix is now released in 5.1.1.

**Comment by adityabandla** [comment]
I am using ANTISMASH v5.1.2 and this issue still seems to persist
